src/file.ts

```diff
@@ -4104,6 +4102,18 @@ class File extends ServiceObject<File> {
       this.storage.retryOptions.autoRetry = false;
     }
   }
+
+  private async getBufferFromReadable(readable: Readable): Promise<Buffer> {
+    readable.once('error', e => {
```

Comment:
There seems to be some difference in how node < 16 handles errors with streams / async iterators. Test cases were failing with Node 12 / 14 until I explicitly handled a stream error. However, node 16 handled it with just the iterator.

Reply:
Broke other things... investigating a better way.

Reply:
Yea, this block handled the error a bit differently - the throw wouldn't populate in `getBufferFromReadable`, but would have a separate context as it is in a callback.

I actually don't think the catch is necessary - `for await (const chunk of readable) {` should populate the throw. E.g. https://nodejs.org/api/stream.html#readablesymbolasynciterator